### ISSUE TITLE
[FP] "Null pointer exception may occur here" after instanceof check

### ISSUE THREAD

**Issue by tom-reinders** [kalessil/phpinspectionsea#1141]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                         |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 5.6                       |

Current behaviour
--

```php
<?php

class Service
{
    /**
     * @param Car $car
     * @return string
     */
    public function getSlogan(Car $car)
    {
        $parts = [
            ($car->getModel() instanceof CarModel) ? $car->getModel()->getName() : '',
            //                                                       ^ alert here
            // ...
        ];

        return implode(' ', $parts);
    }
}

class Car
{
    /**
     * @var CarModel|null
     */
    private $model;

    /**
     * @return CarModel|null
     */
    public function getModel()
    {
        return $this->model;
    }
}

class CarModel
{
    protected $name;

    /**
     * @return string
     */
    public function getName()
    {
        return $this->name;
    }
}

$service = new Service();

$car = new Car();

$service->getSlogan($car);
```

Expected behaviour
--
There should not be a alert.

Environment details
---
PhpStorm 2018.3.1
Build #PS-183.4588.67, built on December 5, 2018
JRE: 1.8.0_152-release-1343-b16 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.19.6-200.fc28.x86_64
Fedora 28

**Comment by andreasschroth** [comment]
Might be related to https://github.com/kalessil/phpinspectionsea/issues/1139 ?

**Comment by kalessil** [comment]
@andreasschroth, this one is similar, but the context is different (ternary here, binary operation in the referenced one).

**Comment by kalessil** [comment]
Fixed!
